File: src/main/java/dev/iancmd/operationrelife/blocks/world/UpperSurfaceIce.java
package dev.iancmd.operationrelife.blocks.world;

import net.minecraft.core.BlockPos;
import net.minecraft.util.RandomSource;
import net.minecraft.world.level.LevelReader;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.material.Material;

public class UpperSurfaceIce extends AbstractIceBlock {
  public UpperSurfaceIce() {
    super(Properties.of(Material.STONE));
  }
  @Override
  public int getExpDrop(BlockState state, LevelReader level, RandomSource randomSource, BlockPos pos, int fortuneLevel, int silkTouchLevel) {
    return 100;
  }
}

File: src/main/kotlin/dev/iancmd/operationrelife/block/world/UpperSurfaceIce.kt
package dev.iancmd.operationrelife.block.world

import net.minecraft.core.BlockPos
import net.minecraft.util.RandomSource
import net.minecraft.world.level.LevelReader
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.material.MapColor

class UpperSurfaceIce : AbstractIceBlock(Properties.of().mapColor(MapColor.ICE)) {
  override fun getExpDrop(
    state: BlockState,
    level: LevelReader,
    randomSource: RandomSource,
    pos: BlockPos,
    fortuneLevel: Int,
    silkTouchLevel: Int
  ): Int {
    return 100
  }
}
